trunk/SUAVE/Input_Output/OpenVSP/vsp_read_prop.py

```diff
@@ -0,0 +1,162 @@
+
+# ----------------------------------------------------------------------
+#  Imports
+# ----------------------------------------------------------------------
+
+import SUAVE
+from SUAVE.Core import Units, Data
+from SUAVE.Input_Output.OpenVSP import get_vsp_areas
+from SUAVE.Components.Wings.Airfoils.Airfoil import Airfoil 
+import vsp_g as vsp
+import numpy as np
+
+
+## @ingroup Input_Output-OpenVSP
+def vsp_read_prop(prop_id, units='SI'): 	
+	"""This reads an OpenVSP propeller geometry and writes it to a SUAVE propeller format.
+
+	Assumptions:
+	Written for OpenVSP 3.16.1
+
+	Source:
+	N/A
+
+	Inputs:
+	1. A tag for an XML file in format .vsp3.
+	2. Units set to 'SI' (default) or 'Imperial'
+
+	Outputs:
+	Writes SUAVE propeller with these geometries from VSP:    (all defaults are SI, but user may specify Imperial)
+		Propellers.Propeller.
+			location[X,Y,Z]                            [radians]
+			rotation[X,Y,Z]                            [radians]
+			prop_attributes.tip_radius                 [m]
+		        prop_attributes.hub_radius                 [m]
+			thrust_angle                               [radians]
+	
+	Note: fills arrays with parametric curve points for:
+	        twists
+		chords
+		skews
+		rakes
+		sweeps
+		
+	Properties Used:
+	N/A
+	"""  	
+
+	prop = SUAVE.Components.Energy.Converters.Propeller()
+	
+	if units == 'SI':
+		units = Units.meter 
+	else:
+		units = Units.foot	
+	
+	if vsp.GetGeomName(prop_id): # Mostly relevant for eVTOLs with > 1 propeller.
+		prop.tag = vsp.GetGeomName(prop_id)
+	else: 
+		prop.tag = 'PropGeom'	
+	
+	prop.prop_attributes.number_blades = vsp.GetParmVal(prop_id, 'NumBlade', 'Design')
+	tip_radius = (vsp.GetParmVal(prop_id, 'Diameter', 'Design')/2.) * units
+	prop.prop_attributes.tip_radius = tip_radius
+	prop.prop_attributes.hub_radius = vsp.GetParmVal(prop_id, 'RadiusFrac', 'XSec_0') * tip_radius	
+	
+	prop.location[0] = vsp.GetParmVal(prop_id, 'X_Rel_Location', 'XForm') * units
+	prop.location[1] = vsp.GetParmVal(prop_id, 'Y_Rel_Location', 'XForm') * units
+	prop.location[2] = vsp.GetParmVal(prop_id, 'Z_Rel_Location', 'XForm') * units
+	prop.rotation[0] = vsp.GetParmVal(prop_id, 'X_Rel_Rotation', 'XForm') * Units.deg
+	prop.rotation[1] = vsp.GetParmVal(prop_id, 'Y_Rel_Rotation', 'XForm') * Units.deg
+	prop.rotation[2] = vsp.GetParmVal(prop_id, 'Z_Rel_Rotation', 'XForm') * Units.deg
+	
+	prop.thrust_angle = prop.rotation[1]			# Y-rotation for thrust angle.
+	
+	xsecsurf_id = vsp.GetXSecSurf(prop_id, 0)
+	
+	curve_type = {0:'linear',1:'spline',2:'Bezier_cubic'}
+	
+	# -------------
+	# Blade geometry
+	# -------------	
+	
+	# Chord
+	chord_curve = curve_type[int(vsp.GetParmVal(prop_id, 'CrvType', 'Chord'))]
+	chord_split_point = vsp.GetParmVal(prop_id, 'SplitPt', 'Chord')
+	chords = []
+	chords_rad = []  							# This is r/R value.
+	chords_num = 50  							# HARDCODED, see break below.
+	for ii in xrange(chords_num):						# Future API call goes for Pcurve chord number goes here.
```

Comment:
What happens when you try to access a chord that doesnt exist?

If OpenVSP has 4 chord stations and you try to access 50 does it give you anything back?